src/DateRangePicker.ts

```diff
@@ -1,26 +1,423 @@
-import { html, css, LitElement, property } from 'lit-element';
+import {
+  html,
+  svg,
+  css,
+  LitElement,
+  TemplateResult,
+  SVGTemplateResult,
+  property,
+  query,
+} from 'lit-element';
+import dayjs from 'dayjs/esm/index.js';
+
+// these values can be overridden via the component's HTML (camelCased) attributes
+const WIDTH = 180;
+const HEIGHT = 40;
+const SLIDER_WIDTH = 10;
+const TOOLTIP_WIDTH = 125;
+const TOOLTIP_HEIGHT = 30;
+const DATE_FORMAT = 'M/D/YYYY';
+
+// these constants are not set up to be overridden
+const DATE_PARSING_ERROR = -1;
+const SLIDER_CORNER_SIZE = 4;
+
+// these CSS custom props can be overridden from the HTML that is invoking this component
+const sliderFill = 'var(--dateRangePickerSliderFill, #4B65FE)';
+const selectedRangeFill = 'var(--dateRangePickerSelectedRangeFill, #DBE0FF)';
+const barIncludedFill = 'var(--dateRangePickerBarIncludedFill, #2C2C2C)';
+const barExcludedFill = 'var(--dateRangePickerBarExcludedFill, #CCCCCC)';
+const inputBorder = css`var(--dateRangePickerInputBorder, 0.5px solid #2C2C2C)`;
+const inputWidth = css`var(--dateRangePickerInputWidth, 70px)`;
+const inputFontSize = css`var(--dateRangePickerInputFontSize, 1.2rem)`;
+const tooltipBackgroundColor = css`var(--dateRangePickerTooltipBackgroundColor, #2C2C2C)`;
+const tooltipTextColor = css`var(--dateRangePickerTooltipTextColor, #FFFFFF)`;
+const tooltipFontSize = css`var(--dateRangePickerTooltipFontSize, 1.1rem)`;
+
+type SliderIds = 'slider-min' | 'slider-max';
+type InputIds = 'date-min' | 'date-max';
+
+interface HistogramInputData {
+  minDate: string;
+  maxDate: string;
+  bins: number[];
+}
+
+interface HistogramItem {
+  value: number;
+  height: number;
+  binStart: string;
+  binEnd: string;
+}
 
 export class DateRangePicker extends LitElement {
+  /* eslint-disable lines-between-class-members */
+
+  // these properties are intended to be passed in as attributes
+  @property({ type: Number }) width = WIDTH;
+  @property({ type: Number }) height = HEIGHT;
```

Comment:
Is `width` and `height` the size of the component itself? If so, instead of having the consumer pass it in as a property, I would let it lay out the component like normal with CSS and use a [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) inside the component to determine its own size.

Reply:
Hmm, that's interesting. I'll look into this.